modules/wowchemy-plugin-reveal/assets/js/wowchemy-slides.js

```diff
@@ -9,105 +9,120 @@
   global RevealMarkdown, RevealSearch, RevealNotes, RevealMath, RevealZoom, Reveal, mermaid, RevealMenu
 */
 
-import * as params from '@params';
-
-// Enable core slide features.
-var enabledPlugins = [RevealMarkdown, RevealSearch, RevealNotes, RevealMath.KaTeX, RevealZoom];
-
-const isObject = function (o) {
-  return o === Object(o) && !isArray(o) && typeof o !== 'function';
-};
-
-const isArray = function (a) {
-  return Array.isArray(a);
-};
-
-const toCamelCase = function (s) {
-  return s.replace(/([-_][a-z])/gi, function (term) {
-    return term.toUpperCase().replace('-', '').replace('_', '');
-  });
-};
-
-const keysToCamelCase = function (o) {
-  if (isObject(o)) {
-    const n = {};
-
-    Object.keys(o).forEach(function (k) {
-      n[toCamelCase(k)] = keysToCamelCase(o[k]);
-    });
-
-    return n;
-  } else if (isArray(o)) {
-    return o.map(function (i) {
-      return keysToCamelCase(i);
-    });
-  }
-
-  return o;
-};
-
-// reveal configurations can be included in front matter under slides.reveal
-var pluginOptions = {};
-if (typeof params.slides.reveal_options !== 'undefined') {
-  pluginOptions = params.slides.reveal_options;
-}
-
-pluginOptions = keysToCamelCase(pluginOptions);
-
-//enable menu by default if not set
-if (typeof pluginOptions.menu_enabled === 'undefined') {
-  pluginOptions.menu_enabled = true;
-}
-
-// configure menu if enabled
-if (pluginOptions.menu_enabled) {
-  enabledPlugins.push(RevealMenu);
-}
-
-pluginOptions['plugins'] = enabledPlugins;
-
-Reveal.initialize(pluginOptions);
-
-// Disable Mermaid by default.
-if (typeof params.slides.diagram === 'undefined') {
-  params.slides.diagram = false;
-}
+export const wowchemySlides = {
```

Comment:
GitHub shows ~130 lines of changes to this file, which may take time for the community to review, especially without much explanation of the changes. Can you clarify what all these changes are?

Reply:
The only two changes are

1. All the code is wrapped inside a function `init` which is then exported as the object {init}. That's the reason for the big diff. `params.slides`, previously imported, is replaced by the argument `slideParams` of `init`. 

2. The Reveal math plugin is set based on the value of the new option `math_renderer`

Reply:
Maybe the [diff without whitespace](https://github.com/wowchemy/wowchemy-hugo-themes/pull/2865/files/?w=1) is more helpful here